src/parsers/installrdf.js

```diff
@@ -15,12 +15,23 @@ export default class InstallRdfParser {
   }
 
   getMetadata() {
-    return Promise.resolve({
-      guid: this._getGUID(),
-      name: this._getName(),
-      type: this._getAddonType(),
-      version: this._getVersion(),
-      restartless: this._getIsBootstrapped(),
+    return new Promise((resolve, reject) => {
+      const guid = this._getGUID();
+      const name = this._getName();
+      const type = this._getAddonType();
+      const version = this._getVersion();
+      const restartless = this._getIsBootstrapped();
+      if (guid === null || name === null || version === null) {
+        return reject(new Error(oneLine`Cannot get metadata,
+                                  version, name, or id tag is missing`));
```

Comment:
You can put the whole string on the next line to make it easier to read, something like:

```javascript
return reject(new Error(oneLine`
    Cannot get metadata, version, name or id tag is missing`));
```

Also, I'd write the error message a bit differently: `Cannot get metadata, version, name or id tag.` The `is missing` part is a bit confusing imho.